controller/metrics/prometheus.py

```diff
@@ -157,67 +197,62 @@ def _collect_labels_from_manifest(self, manifest: Dict[str, Any]) -> Dict[str, s
+            PrometheusMetricAction.observe,
             resource_request.memory_bytes,
             manifest_labels,
         )
-        self._sessions_gpu_request.manipulate(
-            "observe",
+        self._sessions_gpu_request(
+            PrometheusMetricAction.observe,
             resource_request.gpus,
             manifest_labels,
         )
 
     def _on_stop(self, metric_event: MetricEvent):
         manifest_labels = self._collect_labels_from_manifest(metric_event.session)
-        self._sessions_total_deleted.manipulate("inc", 1, manifest_labels)
+        self._sessions_total_deleted(PrometheusMetricAction.inc, 1, manifest_labels)
 
     def _on_any_status_change(self, metric_event: MetricEvent):
         manifest_labels = self._collect_labels_from_manifest(metric_event.session)
-        old_status = metric_event.old_status
-        new_status = metric_event.session.get("status", {}).get("state")
-        if new_status:
-            new_status = ServerStatusEnum(new_status)
         if (
-            old_status == ServerStatusEnum.Starting
-            and new_status == ServerStatusEnum.Running
+            metric_event.old_status == ServerStatusEnum.Starting
+            and metric_event.status == ServerStatusEnum.Running
         ):
-            started_at_str = metric_event.session.get("metadata", {}).get("creationTimestamp")
-            started_at = parser.isoparse(started_at_str) if started_at_str else None
+            started_at = metric_event.session.get("metadata", {}).get("creationTimestamp")
             if started_at:
                 launch_duration = (metric_event.event_timestamp - started_at).total_seconds()
-                self._sessions_launch_duration.manipulate(
-                    "observe",
+                self._sessions_launch_duration(
+                    PrometheusMetricAction.observe,
                     launch_duration,
                     manifest_labels,
                 )
-        if old_status != new_status:
+        if metric_event.status != metric_event.old_status:
             status_change_labels = {
                 **manifest_labels,
-                "status_from": old_status.value if old_status else "None",
-                "status_to": new_status.value if new_status else "None",
+                "status_from": metric_event.old_status.value if metric_event.old_status else "None",
+                "status_to": metric_event.status.value if metric_event.status else "None",
             }
-            self._sessions_status_changes.manipulate("inc", 1, status_change_labels)
+            self._sessions_status_changes(PrometheusMetricAction.inc, 1, status_change_labels)
 
     def publish(self, metric_event: MetricEvent):
+        """Publishes (i.e. persists) the proper prometheus metrics
+        depending on the old and new statuses of the jupyterserver."""
         old_status = metric_event.old_status
         new_status = metric_event.session.get("status", {}).get("state")
```

Comment:
```suggestion
        new_status = metric_event.status
```
?

Reply:
Not sure about this one, `metric_event.session.get("status", {}).get("state")` might be something else. Ignore if it is